File: frontend/clean_hsl.py
import os
import re

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Matches: [hsl(var(--card))] -> card
    # Matches: bg-[hsl(var(--card))] -> bg-card
    # Matches: text-[hsl(var(--muted-foreground))] -> text-muted-foreground
    
    new_content = re.sub(r'\[hsl\(var\(--([a-zA-Z0-9-]+)\)\)\]', r'\1', content)
    
    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated {filepath}")

def main():
    for root, dirs, files in os.walk('src'):
        for file in files:
            if file.endswith(('.tsx', '.ts', '.css')):
                process_file(os.path.join(root, file))

if __name__ == "__main__":
    main()

File: frontend/clean_hsl.mjs
import fs from 'fs';
import path from 'path';

function processFile(filepath) {
    let content = fs.readFileSync(filepath, 'utf-8');
    
    // Matches: [hsl(var(--card))] -> card
    // Matches: bg-[hsl(var(--card))] -> bg-card
    // Matches: text-[hsl(var(--muted-foreground))] -> text-muted-foreground
    const newContent = content.replace(/\[hsl\(var\(--([a-zA-Z0-9-]+)\)\)\]/g, '$1');
    
    if (newContent !== content) {
        fs.writeFileSync(filepath, newContent, 'utf-8');
        console.log(`Updated ${filepath}`);
    }
}

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walkDir(fullPath);
        } else if (fullPath.endsWith('.tsx') || fullPath.endsWith('.ts') || fullPath.endsWith('.css')) {
            processFile(fullPath);
        }
    }
}

walkDir('src');
